src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ChannelDispatcherCollection.cs

```diff
@@ -16,7 +16,15 @@ internal ChannelDispatcherCollection(ServiceHostBase service, object syncRoot)
 
         protected override void ClearItems()
         {
-            throw new PlatformNotSupportedException();
+            ChannelDispatcherBase[] array = new ChannelDispatcherBase[Count];
+            CopyTo(array, 0);
+            base.ClearItems();
```

Comment:
ClearItems should only get called on service shutdown. There's still some cleanup work needed with cleaning everything up with the Kestrel based host is shutting down so this will tie into that. Basically the logic in ServiceHost.Close needs to be moved to the shutdown handler registered with ASP.NET Core.